compiler/passes/docs.cpp

```diff
@@ -592,3 +598,103 @@ static inline std::string ltrim(std::string s) {
   s.erase(s.begin(), std::find_if(s.begin(), s.end(), std::not1(std::ptr_fun<int, int>(std::isspace))));
   return s;
 }
+
+/*
+ * Return true if 's' is empty or only has whitespace characters.
+ */
+static inline bool isEmpty(std::string s) {
+  return s.end() == std::find_if(s.begin(), s.end(), std::not1(std::ptr_fun<int, int>(std::isspace)));
+}
+
+/*
+ * Erase 'count' number of characters from beginning of each line in 's'. Just
+ * ltrim() the first line, though.
+ */
+static std::string erase(std::string s, int count) {
+  std::stringstream sStream(s);
+  std::string line;
+  bool first = true;
+  std::string result = std::string("");
+  while (std::getline(sStream, line)) {
+    if (first) {
+      result += ltrim(line);
+      result += std::string("\n");
+      first = false;
+      continue;
+    }
+    line.erase(line.begin(), line.begin() + count);
+    result += line;
+    result += std::string("\n");
+  }
+  return result;
+}
+
+/*
+ * Returns first non empty line of the string. "Empty lines" are those with no
+ * characters or only whitespace characters.
+ */
+static std::string firstNonEmptyLine(std::string s) {
+  std::stringstream sStream(s);
+  std::string line;
+  std::string result;
+  while (std::getline(sStream, line)) {
+    if (!isEmpty(line)) {
+      result = line;
+      break;
+    }
+  }
+  return result;
+}
+
+/*
+ * Iterate through string, skipping the first line, finding the minimum amount
+ * of whitespace before each line.
+ *
+ * FIXME: Find minimum prefix also if every single line begins with
+ *        "\s+*\s". (thomasvandoren, 2015-02-04)
+ */
+static int minimumPrefix(std::string s) {
+  std::stringstream sStream(s);
+  std::string line;
+  bool first = true;
+  int minPrefix = INT_MAX;
+  int currentPrefix;
+  while (std::getline(sStream, line)) {
+    // Skip the first line. It is a special case that often has been trimmed to
+    // some extent.
+    if (first) {
+      first = false;
+      continue;
+    }
+
+    // If line only contains blanks, do not include it in this
+    // computation. Especially in the case that the string is empty.
+    if (isEmpty(line)) {
+      continue;
+    }
+
+    // Find the first non-space character. Record if it is the new minimum.
+    currentPrefix = std::find_if(line.begin(), line.end(), std::not1(std::ptr_fun<int, int>(std::isspace))) - line.begin();
+    if (currentPrefix < minPrefix) {
+      minPrefix = currentPrefix;
+    }
+  }
+  return minPrefix;
+}
+
+/*
+ * Iterate through all lines of s. Print tabs before each line, ltrim the
+ * lines, and print them.
+ */
+void ltrimAndPrintLines(std::string s, std::ofstream *file) {
+  int minPrefix = minimumPrefix(s);
+  std::string trimmedS = erase(s, minPrefix);
+
+  std::stringstream sStream(trimmedS);
+  std::string line;
+  while (std::getline(sStream, line)) {
+    printTabs(file);
```

Comment:
Both. If the line is empty, or even if it has a bunch of white space (e.g. more than the minimum for the entire comment), it will be trimmed in erase() and the remainder will be printed after tabs here.